### DISCUSSION TITLE
Update bar in multiprocessing for chunked data

### DISCUSSION THREAD

**Discussion by ghost** [tqdm/tqdm#1475] [Q&A]
Hello,

I would like to show the progress bar for chunked data when working with multiprocessing pool.

Here is an example:

```
import numpy as np
import tqdm
import multiprocessing as mtp

def split_list(some_list, size):
    for i in range(0, len(some_list), size):
        yield some_list[i:i + size]

def foo(x):
    return x*x

def main():
    a = np.random.rand(10000)

    with mtp.Pool(4) as pool:
      with tqdm.tqdm(total = len(a)) as p_bar:
        for l in split_list(a, 100):
            output = list(pool.imap_unordered(foo, l))
            p_bar.update(len(l))
            

if __name__ == '__main__':
    main()
```

In this example, the progress bar will only update with every chunk size, not as data is made available,

What I want to achieve is something like this for non-chunked data:

```
import numpy as np
import tqdm
import multiprocessing as mtp

def foo(x):
    return x*x

def main():
    a = np.random.rand(10000)
    with mtp.Pool(4) as pool:
        output = list(tqdm.tqdm(pool.imap(foo, a), total = len(a)))
        
    return output
            
if __name__ == '__main__':
    a = main()

```

This will update the progress bar as data is available. 
Is it possible to do the same with chunked data?

**Comment by WASasquatch** [comment]
You can check out how I did it with my chunk downloader here: https://github.com/civitai/civitai_comfy_nodes/blob/main/CivitAI_Model.py#L193

Basically, you need to pass the instance to the chunk download method.
